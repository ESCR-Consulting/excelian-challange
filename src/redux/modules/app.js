const TOGGLE_DRAWER = 'TOGGLE_DRAWER';

export function handleDrawerToggle(show){
    return {
        type: TOGGLE_DRAWER,
        show
    }
}

const initialState = {
    users: [
        {
            "forename": "Susy",
            "surname": "Cuningham",
            "email": "suzy.cuningham@gmail.com",
            "lastLogin": "1/10/2015"
        },{
            "forename": "Bobby",
            "surname": "Daniels",
            "email": "bobbyD@outlook.com",
            "lastLogin": "1/10/2015"
        },{
            "forename": "John",
            "surname": "Walker",
            "email": "johnnyWalker@blue.com",
            "lastLogin": "1/10/2015"
        },{
            "forename": "Eddy",
            "surname": "Stevens",
            "email": "eStevens@yahoo.com",
            "lastLogin": "1/10/2015"
        },{
            "forename": "Jan",
            "surname": "Williams",
            "email": "jDubz@msn.com",
            "lastLogin": "1/10/2015"
        }
    ],
    mobileOpen: false
};

export default function reducer(state = initialState, action){
    switch (action.type){
        case TOGGLE_DRAWER:
            return {...state, mobileOpen: !action.show}
        default:
            return state;
    }
}
