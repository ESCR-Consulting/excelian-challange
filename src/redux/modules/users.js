const SEARCH_CHANGE = 'SEARCH_CHANGE';
const SORT_USERS = 'SORT_USERS';

export function handleSearchChange(value){
    return {
        type: SEARCH_CHANGE,
        value
    }
}

export function sortUsers(sortDirection){
    return {
        type: SORT_USERS,
        sortDirection
    }
}

export const userList = [
    {
        "name": "Susy Cuningham",
        "email": "suzy.cuningham@gmail.com",
        "lastLogin": "2017-12-10T00:00:00+00:00"
    },{
        "name": "Bobby Daniels",
        "email": "bobbyD@outlook.com",
        "lastLogin": "2017-11-10T00:00:00+00:00"
    },{
        "name": "John Walker",
        "email": "johnnyWalker@blue.com",
        "lastLogin": "2017-06-10T00:00:00+01:00"
    },{
        "name": "Eddy Stevens",
        "email": "eStevens@yahoo.com",
        "lastLogin": "2016-01-10T00:00:00+00:00"
    },{
        "name": "Jan Williams",
        "email": "jDubz@msn.com",
        "lastLogin": "2015-12-31T00:00:00+00:00"
    }
]
const initialState = {
    userList,
    searchValue: '',
    sortDirection: true,
};

export default function reducer(state = initialState, action){
    switch (action.type){
        case SEARCH_CHANGE:
            return {
                ...state,
                searchValue: action.value,
                userList: action.value !== ""
                    ? state.userList.filter(user => user.name.toLowerCase()
                        .match(action.value.toLowerCase()))
                    : userList
            };
        case SORT_USERS:
            return action.sortDirection
                ? {...state, userList: [...state.userList.sort((a,b) =>
                        a.name < b.name
                            ? -1
                            : a.name > b.name
                            ? 1
                            : 0)],
                sortDirection: !action.sortDirection}
                : {...state, userList: [...state.userList.sort((a,b) =>
                        a.name > b.name
                            ? -1
                            : a.name < b.name
                            ? 1
                            : 0)],
                sortDirection: !action.sortDirection}
        default:
            return state;
    }
}
