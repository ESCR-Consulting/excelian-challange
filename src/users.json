{
  "users": [
    {
      "forename": "Susy",
      "surname": "Cuningham",
      "email": "suzy.cuningham@gmail.com",
      "lastLogin": "1/10/2015"
    },{
      "forename": "Bobby",
      "surname": "Daniels",
      "email": "bobbyD@outlook.com",
      "lastLogin": "1/10/2015"
    },{
      "forename": "John",
      "surname": "Walker",
      "email": "johnnyWalker@blue.com",
      "lastLogin": "1/10/2015"
    },{
      "forename": "Eddy",
      "surname": "Stevens",
      "email": "eStevens@yahoo.com",
      "lastLogin": "1/10/2015"
    },{
      "forename": "Jan",
      "surname": "Williams",
      "email": "jDubz@msn.com",
      "lastLogin": "1/10/2015"
    }
  ]
}
