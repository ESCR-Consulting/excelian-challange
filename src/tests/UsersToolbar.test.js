import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import UsersToolbar from '../components/UsersToolbar';
import { sortUsers } from "../redux/modules/users";

const classes = {
    toolbar: '[object Object]'
}
configure({ adapter: new Adapter() });

describe('UsersToolbar', () => {
    it('should render correctly', () => {
        const output = shallow(
            <UsersToolbar classes={classes} sortDirection={true} sortUsers={sortUsers()}/>
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});
