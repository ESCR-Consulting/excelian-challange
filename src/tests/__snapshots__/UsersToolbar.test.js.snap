// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersToolbar should render correctly 1`] = `
<UsersToolbar
  classes={
    Object {
      "toolbar": "UsersToolbar-toolbar-1 [object Object]",
    }
  }
  sortDirection={true}
  sortUsers={
    Object {
      "sortDirection": undefined,
      "type": "SORT_USERS",
    }
  }
/>
`;
